CIPs/cip-34.md

```diff
@@ -0,0 +1,65 @@
+
+## Motivation
+<!--The motivation is critical for CIPs that want to change the Conflux protocol. It should clearly explain why the existing protocol specification is inadequate to address the problem that the CIP solves. CIP submissions without sufficient motivation may be rejected outright.-->
+
+For cross-chain interoperability it is important to implement contract-level light clients so that the status and event on one chain can be automatically verified on the other chain. For example, an Ethereum smart contract is able to verify whether a Bitcoin transaction is confirmed with high confidence, by verifying an SPV proof consisting of a Merkle proof  (showing the target transaction is included in a Bitcoin block) and six block headers of consecutive Bitcoin blocks. Therefore, a blockchain is "readable" by other blockchains where contract-level verification is possible.
+
+Currently, the Conflux confirmation rule is too complicated and computationally heavy. Even the light client version rule requires validation of at least tens of block headers and hence too expensive for contracts. The major reason is that those rules prioritize confirmation latency and user experience. However, for cross-chain interoperation use cases the simplicity of rules takes precedence over low confirmation latency.
+
+Therefore, it would be better if Conflux supports a simple,  contract-friendly verification method, even though it likely results in a greater latency.
+
+## Specification
+<!--The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for any of the current Conflux platforms ([conflux-rust](https://github.com/Conflux-Chain/conflux-rust)).-->
+
+The blocks with significant quality constitute a "slow chain" embedded in the tree-graph structure. More specifically, the timer chain blocks have quality 180 times of target quality, and hence the timer chain has expected block generation time ~90 seconds, which can be safely confirmed with Nakamoto consensus. Thus a Conflux SPV proof consisting of sufficiently many (say, seven or eight) chained timer blocks would be sufficient to provide confidence comparable to six-block confirmation in Bitcoin.
+
+To implement efficient SPV proof with timer blocks, Conflux protocol should change in following two points:
```

Comment:
So the change is basically adding two new fields in each header: the hash of the latest valid timer block, and a Merkle root of all receipts since that header, right?

One concern I have is the size of the receipts Merkle tree. While it can be optimized, the proof size still seems significantly larger than with BTC's SPV proofs.

Reply:
Two new fields: yes.
1. hash reference of the latest valid timer block;
2. a Merkle root of all receipts during two valid timer blocks, but not necessarily *since* that latest valid timer block. 
- The concern is that the uncommitted receipts are continuously updating since that header, whereas the receipts Merkle root up to that latest valid timer block (since the second last valid timer block) is already fixed. Hence by moving the receipts Merkle root one step later we can save the cost of updating the receipts Merkle tree.

Regarding size of the SPV proof, it consists of block headers and a Merkle proof. The number of block headers would be comparable to BTC's SPV proof, i.e. 7~8 vs. 6, since the generation time of timer blocks is less than Bitcoin blocks (~90s vs. ~600s).

The size of Merkle proof mainly depends on the depth of receipt Merkle tree. Currently Conflux timer block has quality 180x of usual, i.e. the expected number of transactions (when fully used) during two timer blocks is 1.5k*180=270k, and w.h.p. bounded by 1.5k*(180±14) ≤ 291k. In comparison, every Bitcoin block contains up to ~4k transactions. Therefore, the depth of receipt Merkle tree is ~18.15 vs. ~12, only increased by 51.25%.

Reply:
Thanks for the explanation. I suggest we move the discussion to #36.